src/nvim/event/pty_process.h

```diff
@@ -13,6 +18,27 @@ typedef struct pty_process {
   int tty_fd;
 } PtyProcess;
 
+# ifdef INCLUDE_GENERATED_DECLARATIONS
+#  include "event/pty_process.h.generated.h"
+# endif
+
+#else
+
+# include "nvim/event/libuv_process.h"
+
+typedef struct pty_process {
+  Process process;
+  char *term_name;
+  uint16_t width, height;
+} PtyProcess;
```

Comment:
I think it would be fine to just `#define PtyProcess LibuvProcess`/`#define pty_process_init libuv_process_init` and use constant values where `term_name`/`width`/`height` are queried. A clean way is to wrap all access to those fields in `pty_process` methods and define versions for `#ifndef FEAT_PTY_PROCESS` that return constants.